Libraries/PeriphDrivers/Include/MAX32572/icc.h

```diff
@@ -0,0 +1,96 @@
+/* ****************************************************************************
+ * Copyright (C) 2023 Maxim Integrated Products, Inc. (now owned by Analog
+ * Devices, Inc.), Analog Devices, Inc. All Rights Reserved. This software
+ * is proprietary and confidential to Analog Devices, Inc. and its licensors.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files(the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL MAXIM INTEGRATED BE LIABLE FOR ANY CLAIM, DAMAGES
+ * OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * Except as contained in this notice, the name of Maxim Integrated
+ * Products, Inc. shall not be used except as stated in the Maxim Integrated
+ * Products, Inc. Branding Policy.
+ *
+ * The mere transfer of this software does not imply any licenses
+ * of trade secrets, proprietary technology, copyrights, patents,
+ * trademarks, maskwork rights, or any other form of intellectual
+ * property whatsoever. Maxim Integrated Products, Inc. retains all
+ * ownership rights.
+ *
+ *************************************************************************** */
+
+/**
+ * @file    icc.h
+ * @brief   Instruction Controller Cache(ICC) function prototypes and data types.
+ */
+
```

Comment:
Does the ME55 support the ICC? I was told that this feature is not supported/was removed.

Or are we adding this register file to support the SFCC? (which I was also not aware this was added by design). Please let me know if this is true and I will make the appropriate CMSIS changes to support the SFCC.

Reply:
As far as I know, the ME55 does not support ICC. As you mentioned, I added ICC files to support SFCC because it essentially does what SFCC wants to do. 
After your feedback, I removed ICC to be compatible with the UserGuide and the design. So, currently there is no connection between SFCC and ICC.

Could you please share your thoughts on this again based on the commit about SFCC?

Reply:
Got it. If that's the case, I will add a `sfcc_regs.h` file for this use case. I will send it to you directly and you can make the appropriate changes in your fork.